cdap-event-reader-spi/src/main/java/io/cdap/cdap/spi/events/StartProgramEventDetails.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Details necessary for starting program.
+ */
+public class StartProgramEventDetails {
+
+  private final String appId;
+  private final String namespaceId;
+  private final String programType;
+  private final String programId;
+  @Nullable
+  private final Map<String, String> args;
+
+  /**
+   * Construct StartProgramEventDetails.
+   *
+   * @param appId AppId of program
+   * @param namespaceId NamespaceId
+   * @param programId ProgramId
+   * @param programType ProgramType
+   * @param args Map of args
+   */
+  public StartProgramEventDetails(String appId, String namespaceId, String programId,
+                                   String programType,
+                                   @Nullable Map<String, String> args) {
```

Comment:
Why allowing null? For collection type, if there is nothing, it's better to use empty collection than null.

Reply:
When GSON deserializes the JSON, a missing field defaults to null. I could change this behavior by adding a custom JsonDeserializer to the GSON instance. Do you think I should implement that and then remove the nullable anotation?

Reply:
I changed the constructor to check for null args and if necessary, create an empty map.